src/main.rs

```diff
@@ -27,38 +25,26 @@ struct Args {
 }
 
 fn main() {
-    fn execute_port(port_address: String, serial_string: String) -> () {
-        let mut port : Box<dyn SerialPort>;
-        loop {
-            if let Ok(p) = serialport::new(&port_address, 115_200).open() {
-                port = p;
-                break;
-            }
-        }
-        let output = serial_string.as_bytes();
-        port.write(output).expect("Write failed!");
-        drop(port);
-    }
-
     let args = Args::parse();
     let hashstring = args.directory + " " + &args.command;
     let color = args.tint;
     let board_size = 32;
-    let mut serial_string = String::from("");
+    let mut serial_string = String::with_capacity(128);
 
-    for byte in [
-        Sha256::new().chain_update(&hashstring).finalize().last().unwrap(),
-        Md4::new().chain_update(&hashstring).finalize().last().unwrap(),
-        Tiger::new().chain_update(&hashstring).finalize().last().unwrap(),
-    ] {
-        let value = byte % &board_size;
-        serial_string.push_str(&color.as_str());
+    let mut hasher = DefaultHasher::default();
+    hasher.write(hashstring.as_bytes());
+    let hashsum = hasher.finish();
+
+    for offset in 0..3 {
+        let value = yhash(hashsum + offset) % board_size;
```

Comment:
the `yhash()` function returns wildly different values for sequential inputs, so we don't really need to compute a different expensive and large hash just to throw everything but the last 8 bits away for our three locations. And you can change the '3' in the loop range there to get more random but repeatable values, in case you want to use more than three lights.